internal/coreinternal/golden/sort_attributes.go

```diff
@@ -0,0 +1,96 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package golden
+
+import (
+	"fmt"
+	"sort"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+)
+
+// sorts all Resource Metrics attributes and Datapoint Slice metric attributes
+func SortAllAttributes(ms pmetric.Metrics) {
```

Comment:
```suggestion
func sortMetrics(ms pmetric.Metrics) {
```
A few thoughts on this:
- Let's not export this, since only this package will use this, for now at least.
- We should include `Metrics` in the name, since we may need to do the same with logs and traces later.